src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -94,16 +95,29 @@
   "The list of states a certificate may be in."
   (schema/enum "requested" "signed" "revoked"))
 
+(def CertificateDetails
+  "CA and client certificate details; notAfter, notBefore and serial values."
+  {:not_after     schema/Str
+   :not_before    schema/Str
+   :serial_number schema/Int})
+
 (def CertificateStatusResult
   "Various information about the state of a certificate or
    certificate request that is provided by the certificate
    status endpoint."
-  {:name              schema/Str
-   :state             CertificateState
-   :dns_alt_names     [schema/Str]
-   :subject_alt_names [schema/Str]
-   :fingerprint       schema/Str
-   :fingerprints      {schema/Keyword schema/Str}})
+  (let [base {:name              schema/Str
+              :state             CertificateState
+              :dns_alt_names     [schema/Str]
+              :subject_alt_names [schema/Str]
+              :fingerprint       schema/Str
+              :fingerprints      {schema/Keyword schema/Str}}]
+    ;; The map should either have all of the CertificateDetails keys or none of
+    ;; them
+    (schema/either
```

Comment:
iirc, `schema/either` is deprecated in favor of `schema/conditional` or `schema/if` which would require a predicate to be written to test if a CertDetail key existed and if so use the merge keys, otherwise use the "base" keys. I'll double check that.

Reply:
Yes, I think that's correct. I'll change it.